EqSat/src/UnionFind.cpp

```diff
@@ -24,12 +26,37 @@ Id UnionFind::find(Id id) const
     return id;
 }
 
+Id UnionFind::find(Id id)
+{
+    LUAU_ASSERT(size_t(id) < parents.size());
+
+    // An e-class id 𝑎 is canonical iff find(𝑎) = 𝑎.
+    if (id != parents[size_t(id)])
+        // Note: we don't update the ranks here since a rank
+        // represents the upper bound on the maximum depth of a tree
+        parents[size_t(id)] = find(parents[size_t(id)]);
+    
+    return parents[size_t(id)];
+}
+
 void UnionFind::merge(Id a, Id b)
 {
     LUAU_ASSERT(size_t(a) < parents.size());
     LUAU_ASSERT(size_t(b) < parents.size());
```

Comment:
Since we're calling `find(id)` which does this assertion, we can probably delete these two.

```suggestion
```